FoodBookApp/UI/Views/SpotDetail/SpotDetailView.swift

```diff
@@ -0,0 +1,168 @@
+                                            Text(key).font(.system(size: 18))
+                                            HStack {
+                                                Image(systemName: rating >= 0.5 ? "hand.thumbsup" : "hand.thumbsdown")
+                                                    .font(.system(size: 15))
+                                                    .foregroundColor(.gray)
+                                                Text(String(format: "%.0f%%", rating * 100)).bold().foregroundColor(.gray).font(.system(size: 15))
+                                            
+                                            }
+                                        }.frame(maxWidth: 200, alignment: .leading)
+                                            .padding(.horizontal, 15)
+                                        
+                    
+                                        ZStack {
+                                            GeometryReader { geometry in
+                                                Rectangle()
+                                                    .fill(customGray)
+                                                    .frame(width: geometry.size.width, height: 5)
+                                                    .cornerRadius(5)
+                                            }
+                                            
+                                            GeometryReader { geometry in
+                                                Rectangle()
+                                                    .fill(Color.blue)
+                                                    .frame(width: CGFloat(rating) * geometry.size.width, height: 5)
+                                                    .cornerRadius(5)
+                                            }
+                                        }.padding()
+                                    }
+                                    .padding(.vertical, 5)
+                                }
+                            }
+                        }
+                        .padding(.horizontal, 20)
+                        .cornerRadius(12)
+ 
+                        // Leave a review
+                        HStack {
+                            leaveReviewButton
+                        }
+                        .padding(.vertical, 20)
+                    }
+                }
+            }
+            
+        }.task {
+            _ = try? await model.fetchSpot()
+        }
+    }
+        
+    var backButton: some View {
+        Button(action: {
+            print("Redirecting to browse...") // FIXME: Should actually redirect 
+        }) {
+            Image(systemName: "chevron.left")
+                .foregroundColor(.blue)
+                .font(.system(size: 20))
+            Text("Browse")
+                .foregroundColor(.blue)
+                .font(.system(size: 20))
+        }
+    }
+    
+    var seeMoreButton: some View {
```

Comment:
All of these Views should be on the Components folder inside UI:
![Screenshot 2024-02-29 at 11 11 38 AM](https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/77981728/96819531-3e2f-4b9c-bbf0-1eab182717cc)

For further reference check this tutorial: https://www.kodeco.com/books/swiftui-cookbook/v1.0/chapters/8-creating-reusable-swiftui-components